pkgs/applications/science/logic/why3/with_provers.nix

```diff
@@ -0,0 +1,19 @@
+{ stdenv, makeWrapper, why3 }:
+solvers:
```

Comment:
It would be much better to make the `solvers` parameter an argument of the attrset, rather than adding a second functional parameter. For one, I think you have an error in `all-packages.nix` since you only use `callPackage` on the file itself, which (I don't think, but haven't checked) will not work.

Furthermore, by having `solvers` as an ordinary argument in the attrset, it makes performing overrides much more natural, like any other package -- any `packageOverrides` settings just have to modify the `solvers` parameter and they're done, rather than doing a sort of callPackage/import/pass arguments dance.

Reply:
The way I was thinking of it is that this produces the function that you call later. So I can say
```
nix-env -i $(nix-build -E "with import <nixpkgs> {}; why3WithProvers [alt-ergo coq cvc4 metis spass yices z3]")
```

What I have now works for that. I feel like `solvers` shouldn't be an ordinary argument in the attrset because it isn't really a dependency of a derivation, it's wrapping up some derivations. I was trying to emulate `ghc.withPackages` or `python.withPackages`. Do you think it would be better to include it as an argument in the attrset?

Reply:
Ah, I see what you mean. Yeah, in retrospect I think this is probably fine then. I suppose it also avoids any confusing case where you might have `solvers` in scope (e.g. using `newScope`) which would be pretty confusing.